pysb/kappa.py

```diff
@@ -289,6 +268,101 @@ def run_kasim(model, time=10000, points=200, output_dir='.', cleanup=False,
     return output_dict
 
 
+def run_kasa(model, influence_map=False, contact_map=False, output_dir='.',
+             cleanup=False, base_filename=None):
+    """Run KaSa (static analyzer) on the given model.
+
+    Parameters
+    ----------
+    model : pysb.core.Model
+        The model to simulate/analyze using KaSa.
+    influence_map : boolean
+        Whether to compute the influence map.
+    contact_map : boolean
+        Whether to compute the contact map.
+    output_dir : string
+        The subdirectory in which to generate the Kappa (.ka) file for the
+        model and all output files produced by KaSim. Default value is '.'
+        Note that only relative paths can be specified; paths are relative
+        to the directory where the current Python instance is running.
+        If the specified directory does not exist, an Exception is thrown.
+    cleanup : boolean
+        Specifies whether output files produced by KaSim should be deleted
+        after execution is completed. Default value is False.
+    base_filename : The base filename to be used for generation of the Kappa
+        (.ka) file and all output files produced by KaSim. Defaults to a
+        string of the form::
+
+            '%s_%d_%d_temp' % (model.name, program id, random.randint(0,10000))
+
+        The influence map filename appends '_im.dot' to this base filename; the
+        contact map filename appends '_cm.dot'.
+
+    Returns
+    -------
+    A dict with two entries giving the filenames for the files produced
+
+        * output_dict['im'] gives the influence map filename, or None if not
+          produced
+        * output_dict['cm'] gives the contact map filename, or None if not
+          produced
+    """
+
+    gen = KappaGenerator(model)
+
+    if not base_filename:
+        base_filename = '%s/%s_%d_%d_temp' % (output_dir,
```

Comment:
Should probably use `tempfile.mkdtemp` to create a directory without any races, even though the approach here is probably sufficient. See https://github.com/pysb/pysb/pull/180/files#diff-8f7a45525fb86b609c6d0eb1db710faeR104 for how Alex did it in the BNG interface. Note though that the default for BNG is cleanup=True because we read everything most users will want from the cdat/gdat files into memory. Also, why does `output_dir` have to be relative? Some environments (e.g. CGI scripts) may have a problem with that -- they'd want to write somewhere under `/tmp`.